tests/publiccloud/check_registercloudguest.pm

```diff
@@ -26,18 +26,17 @@ my $path = is_sle('=15-SP2') ? '/usr/sbin/' : '';    # 15-SP2 is the oldest vers
 my $regcode_param = (is_byos()) ? "-r " . get_required_var('SCC_REGCODE') : '';
 
 sub run {
-    my ($self, $args) = @_;
-    my ($provider, $instance);
+    my ($self) = @_;
+    my ($instance);
     select_host_console();
 
     $run_count++;
 
     if (get_var('PUBLIC_CLOUD_QAM', 0)) {
-        $instance = $self->{my_instance} = $args->{my_instance};
-        $provider = $self->{provider} = $args->{my_provider};    # required for cleanup
+        $instance = $self->instance;
```

Comment:
why ? now we can always just access it directly `$self->instance`

Reply:
sure. this was just to make the minimal changes to show how it would work.